File: domain/src/main/java/com/chronos/mobile/domain/GithubContributorRepository.kt
package com.chronos.mobile.domain

import com.chronos.mobile.domain.model.GithubContributor
import com.chronos.mobile.domain.result.AppResult

interface GithubContributorRepository {
    suspend fun fetchContributors(
        owner: String,
        repo: String,
        limit: Int,
    ): AppResult<List<GithubContributor>>
}

File: src/lib/domain/interfaces/github-contributor-repository.ts
import type { GithubContributor } from '$lib/models/auth';
import type { AppResult } from '../result';

export interface GithubContributorRepository {
	fetchContributors(
		owner: string,
		repo: string,
		limit: number
	): Promise<AppResult<GithubContributor[]>>;
}
